src/crawlee/storages/_creation_management.py

```diff
@@ -16,8 +17,8 @@
 TResource = TypeVar('TResource', Dataset, KeyValueStore, RequestQueue)
 
 
-_creation_lock = asyncio.Lock()
-"""Lock for storage creation."""
+_creation_locks = WeakKeyDictionary[asyncio.AbstractEventLoop, asyncio.Lock]()
+"""Locks for storage creation (we need a separate lock for every event loop so that tests work as expected)."""
```

Comment:
> we need a separate lock for every event loop so that tests work as expected

So, are the changes in this file (multiple locks instead of just one) here just because of tests?